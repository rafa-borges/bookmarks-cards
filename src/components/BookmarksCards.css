/* BookmarksCards.css */
.list-group {
    margin: 3px;
}
.list-group-item {
    margin: 2px;
    border-width: 2px 0px 2px 0px;
    padding: 0px 6px 0px 8px;
    vertical-align: middle;
    border: none;
    cursor: pointer;
}
.list-group-item:hover {
    border-radius: 4px;
}
.list-group-item.key-link {
    font-weight: bold;
    font-size: 14px;
    min-height: 24px;
    background: #F2CC38;
    color: #1e1e1e;
}
.list-group-item.key-link:hover {
    background-color: #00A082;
    color: #f1f1f1;
}
.list-group-item.standard-link {
    font-size: 13px;
    background: #363636;
    color: #f1f1f1;
}
.list-group-item.standard-link:hover {
    background-color: #00A082;
    color: #F2CC38;
}
.item-icon {
    margin-top: -2px;
    margin-right: 6px;
    height: 16px;
    width: 16px;
}
.card-container {
    column-count: 1;
    column-gap: 12px;
}
.card {
    border-color: #000000;
    margin-bottom: 8px;
    min-width: 250px;
    max-width: 500px;
    background: #363636;
    break-inside: avoid;
}
.card-header {
    background-color: #1e1e1e;
    border-color: #000000;
    color: #00A082;
    font-size: 14px;
    font-weight: bold;
    padding: 6px 6px 6px 10px;
    cursor: pointer;
}
.card-header:hover {
    background-color: #00A082;
    color: #F2CC38;
}
.card-body {
    border-color: #000000;
}
.card-as-overlay {
    margin-top: 6px;
    border-color: #00A082;
    border-width: 2px;
}

.key-link {
    text-align: center;
    max-width: 100%;
    background: #F2CC38;
}

@media (min-width: 530px) {
    .card-container {
        column-count: 2;
    }
}

@media (min-width: 800px) {
    .card-container {
        column-count: 3;
    }
}

@media (min-width: 1100px) {
    .card-container {
        column-count: 4;
    }
}

@media (min-width: 1400px) {
    .card-container {
        column-count: 5;
    }
}

@media (min-width: 1650px) {
    .card-container {
        column-count: 6;
    }
}

@media (min-width: 2000px) {
    .card-container {
        column-count: 7;
    }
}

@media (min-width: 2200px) {
    .card-container {
        column-count: 8;
    }
}
