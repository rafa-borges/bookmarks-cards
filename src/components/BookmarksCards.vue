<template>
  <div>
    <b-card v-for="(card, i) in keyCards(cards)" :key="i" no-body class="key-link">
      <b-list-group horizontal>
        <b-list-group-item v-for="(item, j) in card.items" :key="j" @click="clickOnCardItem(item)" class="flex-fill key-link">
          <b-img v-if="item.icon" :src="item.icon" class="item-icon"/>
          {{ item.name }}
        </b-list-group-item>
      </b-list-group>
    </b-card>
    <div style="margin: 10px"/>
    <b-card-group columns>
      <b-card v-for="(card, i) in standardCards(cards)" :key="i" no-body>
        <b-card-header @click="clickOnCard(card)">{{ card.name }}</b-card-header>
        <b-list-group>
          <b-list-group-item v-for="(item, j) in card.items" :key="j" @click="clickOnCardItem(item)" class="standard-link">
            <b-img v-if="item.icon" :src="item.icon" class="item-icon"/>
            {{ item.name }}
          </b-list-group-item>
        </b-list-group>
      </b-card>
    </b-card-group>
  </div>
</template>

<script lang="ts">
export default {
  props: {
    cards: Array,
    clickOnCard: Function,
    clickOnCardItem: Function
  }, 
  methods: {
    keyCards(cards) {
      return cards.slice(0, 1)
    },
    standardCards(cards) {
      return cards.slice(1)
    }    
  }
}
</script>

<style scoped>
  div.list-group {
    margin: 3px;
  }
  div.list-group-item {
    margin: 2px;
    border: 2px 0px 2px 0px;
    padding: 0px 6px 0px 8px;
    vertical-align: middle;
    border: none;
    cursor: pointer;
  }
  div.list-group-item:hover {
    border-radius: 4px;
  }
  div.list-group-item.key-link {
    font-weight: bold;
    font-size: 14px;
    min-height: 24px;
    background: #F2CC38;
    color: #1e1e1e;
  }
  div.list-group-item.key-link:hover {
    background-color: #00A082;
    color: #f1f1f1;
  }
  div.list-group-item.standard-link {
    font-size: 13px;
    background: #363636;
    color: #f1f1f1;
  }
  div.list-group-item.standard-link:hover {
    background-color: #00A082;
    color: #F2CC38;
  }
  .item-icon {
    margin-top: -2px;
    margin-right: 1px;
    height: 16px;
    width: 16px;
  }
  .card {
    border-color: #000000;
    min-width: 150px;
    max-width: 600px;
    background: #363636;
  }
  .card-header {
    background-color: #1e1e1e;
    border-color: #000000;
    color: #00A082;
    font-size: 14px;
    font-weight: bold;
    padding: 6px 6px 6px 10px;
    cursor: pointer;
  }
  .card-header:hover {
    background-color: #00A082;
    color: #F2CC38;
  }
  .card-body {
    border-color: #000000;
  }
  .card-columns {
    column-count: 1;
  }
  div.key-link {
    text-align: center;
    max-width: 100%;
    background: #F2CC38;
  }

  @media (min-width: 630px) {
    .card-columns {
      column-count: 2;
    }
  }

  @media (min-width: 1000px) {
    .card {
      max-width: 400px;
    }
    .card-columns {
      column-count: 3;
    }
  }

  @media (min-width: 1300px) {
    .card-columns {
      column-count: 4;
    }
  }

  @media (min-width: 1600px) {
    .card-columns {
      column-count: 5;
    }
  }

  @media (min-width: 1900px) {
    .card-columns {
      column-count: 6;
    }
  }

  @media (min-width: 2100px) {
    .card-columns {
      column-count: 7;
    }
  }

  @media (min-width: 2400px) {
    .card-columns {
      column-count: 8;
    }
  }

</style>
