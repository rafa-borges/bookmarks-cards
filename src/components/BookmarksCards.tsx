import "./BookmarksCards.css"

import React, { useEffect, useState } from "react"
import { Card, ListGroup, Image, OverlayTrigger } from "react-bootstrap"
import { BookmarkCard, BookmarkCardItem } from "../types/BookmarkCards.ts"

interface BookmarksCardsProps {
    cards: BookmarkCard[];
    clickOnCard: (card: BookmarkCard) => void;
    clickOnCardItem: (item: BookmarkCardItem) => void;
}

const CardHeader: React.FC<{ card: BookmarkCard, isVisible: boolean, onToggle: () => void, onClick: () => void }> = ({ card, isVisible, onToggle, onClick }) => (
  <Card.Header>
    {card.isNested && <span onClick={onToggle}>{isVisible ? " ➖ " : " ➕ "}</span>}
    <span onClick={onClick}>{card.name}</span>
  </Card.Header>
)

const CardItems: React.FC<{ items: BookmarkCardItem[], onClickItem: (item: BookmarkCardItem) => void }> = ({ items, onClickItem }) => (
  <ListGroup>
    {items.map((item, index) => (
      <ListGroup.Item key={index} onClick={() => onClickItem(item)} className="standard-link">
        {item.icon && <Image src={item.icon} className="item-icon" />}
        {item.name}
      </ListGroup.Item>
    ))}
  </ListGroup>
)

const HiddenCardContent: React.FC<{ index: number, items: BookmarkCardItem[], onClickItem: (item: BookmarkCardItem) => void, width: number | null }> = ({ index, items, onClickItem, width }) => (
  <OverlayTrigger
    trigger={['hover','click']}
    placement="bottom"
    delay={{ show: 500, hide: 2500 }}
    overlay={
      <Card key={`ovl-${index}`} body={false} style={{ width: width ?? 'auto' }} className="card-as-overlay">
        <CardItems items={items} onClickItem={onClickItem} />
      </Card>}>
    <ListGroup>
      <ListGroup.Item className="standard-link">
        {items.map((item, index) => item.icon && <Image key={index} src={item.icon} className="item-icon" />)}
      </ListGroup.Item>
    </ListGroup>
  </OverlayTrigger>
)

const BookmarksCards: React.FC<BookmarksCardsProps> = ({ cards, clickOnCard, clickOnCardItem }) => {
    const keyCards = cards.slice(0, 1)
    const standardCards = cards.slice(1)

    const [visibleCardIndices, setVisibleCardIndices] = useState<boolean[]>(standardCards.map(card => !card.isNested))
    const [listGroupItemWidth, setListGroupItemWidth] = useState<number | null>(null)
    const listGroupItemRef = React.useRef<HTMLDivElement>(null)

    const toggleVisibility = (index: number) => {
      setVisibleCardIndices(prevState => {
        const newState = [...prevState]
        newState[index] = !newState[index]
        return newState
      })
    }

    useEffect(() => {
      if (listGroupItemRef.current) {
        setListGroupItemWidth(listGroupItemRef.current.offsetWidth)
      }
    }, [])

    return (
        <div>
            {keyCards.map((card, i) => (
                <Card key={`key-${i}`} className="key-link">
                    <ListGroup horizontal>
                        {card.items.map((item, j) => (
                            <ListGroup.Item key={j} onClick={() => clickOnCardItem(item)} className="flex-fill key-link">
                                {item.icon && <Image src={item.icon} className="item-icon" />}
                                {item.name}
                            </ListGroup.Item>
                        ))}
                    </ListGroup>
                </Card>
            ))}
            <div style={{ margin: '10px' }} />

            <div className="card-container">
                {standardCards.map((card, i) => (
                  <Card key={`std-${i}`} body={false} ref={listGroupItemRef}>
                    <CardHeader
                      card={card}
                      isVisible={visibleCardIndices[i]}
                      onToggle={() => toggleVisibility(i)}
                      onClick={() => clickOnCard(card)}/>
                    {visibleCardIndices[i] ? (
                      <CardItems items={card.items} onClickItem={clickOnCardItem} />
                    ) : (
                      <HiddenCardContent index={i} items={card.items} onClickItem={clickOnCardItem} width={listGroupItemWidth} />
                    )}
                  </Card>
                ))}
            </div>
        </div>
    );
};

export default BookmarksCards;