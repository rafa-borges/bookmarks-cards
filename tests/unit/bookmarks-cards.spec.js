import { shallowMount, createLocalVue } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'

import BookmarksCards from '@/components/BookmarksCards.vue'
import sample from '@/data/sample-cards.js';

const localVue = createLocalVue()
localVue.use(BootstrapVue)

const cards = []
cards.push(...sample)

describe('BookmarksCards.vue', () => {
  it('renders cards when passed', () => {
    const wrapper = shallowMount(BookmarksCards, {
      localVue,
      propsData: { cards }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
