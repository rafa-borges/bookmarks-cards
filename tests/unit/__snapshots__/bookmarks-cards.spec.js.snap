// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarksCards.vue renders cards when passed 1`] = `
<div>
  <b-card-stub bodytag="div" footertag="div" headertag="div" nobody="true" subtitletag="h6" subtitletextvariant="muted" tag="div" titletag="h4" class="key-link">
    <b-list-group-stub horizontal="true" tag="div">
      <b-list-group-item-stub tag="div" target="_self" class="flex-fill key-link">
        <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.gmail.com/" class="item-icon"></b-img-stub>
        Mail
      </b-list-group-item-stub>
      <b-list-group-item-stub tag="div" target="_self" class="flex-fill key-link">
        <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.jira.com" class="item-icon"></b-img-stub>
        Jira
      </b-list-group-item-stub>
      <b-list-group-item-stub tag="div" target="_self" class="flex-fill key-link">
        <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.stackoverflow.com" class="item-icon"></b-img-stub>
        Stackoverflow
      </b-list-group-item-stub>
      <b-list-group-item-stub tag="div" target="_self" class="flex-fill key-link">
        <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://looker.com" class="item-icon"></b-img-stub>
        Looker
      </b-list-group-item-stub>
    </b-list-group-stub>
  </b-card-stub>
  <div style="margin: 10px;"></div>
  <b-card-group-stub columns="true" tag="div">
    <b-card-stub bodytag="div" footertag="div" headertag="div" nobody="true" subtitletag="h6" subtitletextvariant="muted" tag="div" titletag="h4">
      <b-card-header-stub headertag="div">Some More Links 1</b-card-header-stub>
      <b-list-group-stub tag="div">
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.gmail.com/" class="item-icon"></b-img-stub>
          Mail
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.jira.com" class="item-icon"></b-img-stub>
          Jira
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.stackoverflow.com" class="item-icon"></b-img-stub>
          Stackoverflow
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://looker.com" class="item-icon"></b-img-stub>
          Looker
        </b-list-group-item-stub>
      </b-list-group-stub>
    </b-card-stub>
    <b-card-stub bodytag="div" footertag="div" headertag="div" nobody="true" subtitletag="h6" subtitletextvariant="muted" tag="div" titletag="h4">
      <b-card-header-stub headertag="div">Some More Links 2</b-card-header-stub>
      <b-list-group-stub tag="div">
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.gmail.com/" class="item-icon"></b-img-stub>
          Mail
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.jira.com" class="item-icon"></b-img-stub>
          Jira
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.stackoverflow.com" class="item-icon"></b-img-stub>
          Stackoverflow
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://looker.com" class="item-icon"></b-img-stub>
          Looker
        </b-list-group-item-stub>
      </b-list-group-stub>
    </b-card-stub>
    <b-card-stub bodytag="div" footertag="div" headertag="div" nobody="true" subtitletag="h6" subtitletextvariant="muted" tag="div" titletag="h4">
      <b-card-header-stub headertag="div">Some More Links 3</b-card-header-stub>
      <b-list-group-stub tag="div">
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.gmail.com/" class="item-icon"></b-img-stub>
          Mail
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.jira.com" class="item-icon"></b-img-stub>
          Jira
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.stackoverflow.com" class="item-icon"></b-img-stub>
          Stackoverflow
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://looker.com" class="item-icon"></b-img-stub>
          Looker
        </b-list-group-item-stub>
      </b-list-group-stub>
    </b-card-stub>
    <b-card-stub bodytag="div" footertag="div" headertag="div" nobody="true" subtitletag="h6" subtitletextvariant="muted" tag="div" titletag="h4">
      <b-card-header-stub headertag="div">Some More Links 4</b-card-header-stub>
      <b-list-group-stub tag="div">
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.gmail.com/" class="item-icon"></b-img-stub>
          Mail
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.jira.com" class="item-icon"></b-img-stub>
          Jira
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.stackoverflow.com" class="item-icon"></b-img-stub>
          Stackoverflow
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://looker.com" class="item-icon"></b-img-stub>
          Looker
        </b-list-group-item-stub>
      </b-list-group-stub>
    </b-card-stub>
    <b-card-stub bodytag="div" footertag="div" headertag="div" nobody="true" subtitletag="h6" subtitletextvariant="muted" tag="div" titletag="h4">
      <b-card-header-stub headertag="div">Some More Links 5</b-card-header-stub>
      <b-list-group-stub tag="div">
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.gmail.com/" class="item-icon"></b-img-stub>
          Mail
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.jira.com" class="item-icon"></b-img-stub>
          Jira
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.stackoverflow.com" class="item-icon"></b-img-stub>
          Stackoverflow
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://looker.com" class="item-icon"></b-img-stub>
          Looker
        </b-list-group-item-stub>
      </b-list-group-stub>
    </b-card-stub>
    <b-card-stub bodytag="div" footertag="div" headertag="div" nobody="true" subtitletag="h6" subtitletextvariant="muted" tag="div" titletag="h4">
      <b-card-header-stub headertag="div">Some More Links 6</b-card-header-stub>
      <b-list-group-stub tag="div">
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.gmail.com/" class="item-icon"></b-img-stub>
          Mail
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.jira.com" class="item-icon"></b-img-stub>
          Jira
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.stackoverflow.com" class="item-icon"></b-img-stub>
          Stackoverflow
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://looker.com" class="item-icon"></b-img-stub>
          Looker
        </b-list-group-item-stub>
      </b-list-group-stub>
    </b-card-stub>
    <b-card-stub bodytag="div" footertag="div" headertag="div" nobody="true" subtitletag="h6" subtitletextvariant="muted" tag="div" titletag="h4">
      <b-card-header-stub headertag="div">Some More Links 7</b-card-header-stub>
      <b-list-group-stub tag="div">
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.gmail.com/" class="item-icon"></b-img-stub>
          Mail
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.jira.com" class="item-icon"></b-img-stub>
          Jira
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.stackoverflow.com" class="item-icon"></b-img-stub>
          Stackoverflow
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://looker.com" class="item-icon"></b-img-stub>
          Looker
        </b-list-group-item-stub>
      </b-list-group-stub>
    </b-card-stub>
    <b-card-stub bodytag="div" footertag="div" headertag="div" nobody="true" subtitletag="h6" subtitletextvariant="muted" tag="div" titletag="h4">
      <b-card-header-stub headertag="div">Some More Links 8</b-card-header-stub>
      <b-list-group-stub tag="div">
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.gmail.com/" class="item-icon"></b-img-stub>
          Mail
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.jira.com" class="item-icon"></b-img-stub>
          Jira
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.stackoverflow.com" class="item-icon"></b-img-stub>
          Stackoverflow
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://looker.com" class="item-icon"></b-img-stub>
          Looker
        </b-list-group-item-stub>
      </b-list-group-stub>
    </b-card-stub>
    <b-card-stub bodytag="div" footertag="div" headertag="div" nobody="true" subtitletag="h6" subtitletextvariant="muted" tag="div" titletag="h4">
      <b-card-header-stub headertag="div">Some More Links 9</b-card-header-stub>
      <b-list-group-stub tag="div">
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.gmail.com/" class="item-icon"></b-img-stub>
          Mail
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.jira.com" class="item-icon"></b-img-stub>
          Jira
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://www.stackoverflow.com" class="item-icon"></b-img-stub>
          Stackoverflow
        </b-list-group-item-stub>
        <b-list-group-item-stub tag="div" target="_self" class="standard-link">
          <b-img-stub blankcolor="transparent" src="https://www.google.com/s2/favicons?domain=https://looker.com" class="item-icon"></b-img-stub>
          Looker
        </b-list-group-item-stub>
      </b-list-group-stub>
    </b-card-stub>
  </b-card-group-stub>
</div>
`;
